### PR TITLE
Use JSON for the network file

### PR DESCRIPTION
**Pull request by dhdaines**
- Switch to using compressed JSON for the network as well as the langs file
- Cease to use `JSON` and `PKL` in the constant names in `g2p.mappings.langs` in case we decide to change formats again
- Remove curiously specific `networkx` dependency

### CODE REVIEW

File: g2p/mappings/langs/utils.py [hunk 1]
```diff
@@ -169,22 +169,23 @@ def cache_langs(
...
+    with gzip.GzipFile(path, "wb", mtime=0) as zipfile:
+        zipfile.write(
+            json.dumps(
+                data,
                 separators=(",", ":"),
                 ensure_ascii=False,
                 sort_keys=True,
-            )
-    return langs
+            ).encode("utf-8")
```
Comment (joanise) [inline g2p/mappings/langs/utils.py:187]: Wow, haven't seen the use of `encode()` in a long time! I kinda forget that exists, these days...

File: g2p/mappings/langs/utils.py [hunk 2]
```diff
@@ -169,22 +169,23 @@ def cache_langs(
...
 
-    with open(network_path, "wb") as f:
-        write_gpickle(lang_network, f, protocol=4)
 
-    with gzip.GzipFile(langs_path, "wb", mtime=0) as zipfile_raw:
-        with io.TextIOWrapper(zipfile_raw, encoding="utf-8") as zipfile:  # type: ignore
-            json.dump(
-                langs,
-                zipfile,
+def write_json_gz(path: str, data: str):
```
Comment (joanise) [inline g2p/mappings/langs/utils.py:179]: OK, this was worth factoring out, I now agree. 😄

File: requirements/requirements.txt [hunk 3]
```diff
@@ -6,7 +6,7 @@ eventlet>=0.33.0
 flask-cors>=3.0.9
 flask-talisman>=0.7.0
 flask-restful>=0.3.9
-networkx>=2.5,<=2.8.4
+networkx>=3.1
```
Comment (joanise) [inline requirements/requirements.txt]: This will break on Python 3.7.

Comment (joanise) [reply requirements/requirements.txt]: The most recent release of `networkx` for Python 3.7 is 2.6.3... ☹️

Comment (joanise) [reply requirements/requirements.txt]: But the unit tests pass with `>=2.6.3` so we don't need to ask for 3.1

**Review by roedoejet — APPROVED** [review]
Looks great to me, thanks @dhdaines !!

**Review by joanise — CHANGES_REQUESTED** [review]
I'm glad we have good unit tests. This is good. We just have to keep networkx compatible with Python 3.7 and we can merge this.

**Comment by joanise** [comment]
And I should have hit that refresh button GitHub was displaying, you've already made the changes! 😄

**Review by joanise — APPROVED** [review]

**Pull request merged**